src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/probation/ProbationMergeEventProcessor.kt

```diff
@@ -18,7 +19,7 @@ class ProbationMergeEventProcessor(
   val mergeService: MergeService,
 ) : BaseProbationEventProcessor() {
 
-  fun processEvent(domainEvent: DomainEvent) {
+  fun processEvent(domainEvent: DomainEvent) = runBlocking {
```

Comment:
I know that you have refactored this since, but I wanted to explain what I meant yesterday.

This is one of the downsides of using inheritance rather than composition - the move to coroutines is required in one of the subclasses but is forced on all of them. As a result, changes have to be made to classes which are not related to the work in hand and are (possibly) not getting any benefits from the change. This is a sign that there is too much coupling in the codebase.

I'm sure you have heard "favour composition over inheritance" before. I personally have found inheritance to be useful in test classes, and I think this is because they very rarely change functionality, just add or remove test methods, so refactors which affect classes which you don't need to change are unlikely. Inheritance does have its place, especially in polymorphism.

Reply:
Yeah I agree, i think its one to sort our in a future PR as this is getting quite big now